renderdoc/driver/vulkan/vk_debug.cpp

```diff
@@ -2741,24 +2741,43 @@ void VulkanDebugManager::ReplaceResource(ResourceId from, ResourceId to)
 
     if(refdShader)
     {
-      VkGraphicsPipelineCreateInfo pipeCreateInfo;
-      MakeGraphicsPipelineInfo(pipeCreateInfo, it->first);
+      VkPipeline pipe = VK_NULL_HANDLE;
+      const VulkanCreationInfo::Pipeline &pipeInfo = m_pDriver->m_CreationInfo.m_Pipeline[it->first];
+      if(pipeInfo.renderpass.id)    // check if this is a graphics or compute pipeline
```

Comment:
Don't check the `.id` member of a ResourceId() directly, I try to keep the type opaque as much as possible so the `.id` is an "implementation detail". Instead compare like `pipeInfo.renderpass != ResourceId()`.